core/common/test/LocalDateRangeTest.kt

```diff
@@ -0,0 +1,283 @@
+                    (Jan_24_2000 downTo Jan_01_2000).step(step, DateTimeUnit.DAY).random(actualRand)
+                )
+            }
+        }
+        repeat(20) {
+            (Jan_01_2000..Jan_24_2000).step(5, DateTimeUnit.DAY).let { assertContains(it, it.random()) }
+        }
+    }
+
+    @Test
+    fun first() {
+        assertEquals((Jan_01_2000..Jan_01_2000).first(), Jan_01_2000)
+        assertEquals((Jan_01_2000 downTo Jan_01_2000).first(), Jan_01_2000)
+        assertEquals((Jan_01_2000..Jan_05_2000).first(), Jan_01_2000)
+        assertEquals((Jan_05_2000 downTo Jan_01_2000).first(), Jan_05_2000)
+        assertFailsWith<NoSuchElementException> { (Jan_02_2000..Jan_01_2000).first() }
+        assertFailsWith<NoSuchElementException> { (Jan_01_2000 downTo Jan_02_2000).first() }
+    }
+
+    @Test
+    fun last() {
+        assertEquals((Jan_01_2000..Jan_01_2000).last(), Jan_01_2000)
+        assertEquals((Jan_01_2000 downTo Jan_01_2000).last(), Jan_01_2000)
+        assertEquals((Jan_01_2000..Jan_05_2000).last(), Jan_05_2000)
+        assertEquals((Jan_05_2000 downTo Jan_01_2000).last(), Jan_01_2000)
+        assertEquals((Jan_01_2000..Jan_06_2000).step(2, DateTimeUnit.DAY).last(), Jan_05_2000)
+        assertEquals((Jan_06_2000 downTo Jan_01_2000).step(2, DateTimeUnit.DAY).last(), Jan_02_2000)
+        assertFailsWith<NoSuchElementException> { (Jan_02_2000..Jan_01_2000).last() }
+        assertFailsWith<NoSuchElementException> { (Jan_01_2000 downTo Jan_02_2000).last() }
+    }
+
+    @Test
+    fun firstOrNull() {
+        assertEquals((Jan_01_2000..Jan_01_2000).firstOrNull(), Jan_01_2000)
+        assertEquals((Jan_01_2000 downTo Jan_01_2000).firstOrNull(), Jan_01_2000)
+        assertEquals((Jan_01_2000..Jan_05_2000).firstOrNull(), Jan_01_2000)
+        assertEquals((Jan_05_2000 downTo Jan_01_2000).firstOrNull(), Jan_05_2000)
+        assertNull( (Jan_02_2000..Jan_01_2000).firstOrNull() )
+        assertNull( (Jan_01_2000 downTo Jan_02_2000).firstOrNull() )
+    }
+
+    @Test
+    fun lastOrNull() {
+        assertEquals((Jan_01_2000..Jan_01_2000).lastOrNull(), Jan_01_2000)
+        assertEquals((Jan_01_2000 downTo Jan_01_2000).lastOrNull(), Jan_01_2000)
+        assertEquals((Jan_01_2000..Jan_05_2000).lastOrNull(), Jan_05_2000)
+        assertEquals((Jan_05_2000 downTo Jan_01_2000).lastOrNull(), Jan_01_2000)
+        assertNull( (Jan_02_2000..Jan_01_2000).lastOrNull() )
+        assertNull( (Jan_01_2000 downTo Jan_02_2000).lastOrNull() )
+    }
+
+    @Test
+    fun reversed() {
+        assertContentEquals(
+            Jan_05_2000 downTo Jan_01_2000,
+            (Jan_01_2000..Jan_05_2000).reversed()
+        )
+        assertContentEquals(
+            Jan_01_2000..Jan_05_2000,
+            (Jan_05_2000 downTo Jan_01_2000).reversed()
+        )
+        assertContentEquals(
+            Jan_01_2000..Jan_01_2000,
+            (Jan_01_2000..Jan_01_2000).reversed()
+        )
```

Comment:
I believe it should check more stricter - for equality, not for content equality